lib/hammer_cli/options/option_collector.rb

```diff
@@ -0,0 +1,22 @@
+module HammerCLI
+  module Options
+    class OptionCollector
+      attr_accessor :option_sources
+
+      def initialize(recognised_options, option_sources)
+        @recognised_options = recognised_options
+        @option_sources = option_sources
+      end
+
+      def all_options
+        @all_options ||= @option_sources.inject({}) do |all_options, source|
+          source.get_options(@recognised_options, all_options)
```

Comment:
I've seen the `result` is evaluated in `SavedDefaults` option source but could each option source just return its values and get them merged together in order in which the sources are stored in the collector?